crates/editor/src/element.rs

```diff
@@ -647,7 +647,9 @@ impl EditorElement {
             if !text_hitbox.is_hovered(cx) || editor.read_only(cx) {
                 return;
             }
-
+            if !EditorSettings::get_global(cx).linux_middle_click_paste {
+                return;
+            }
```

Comment:
I think this should go into the `cfg(target_os = "linux")` blog, right? And then we can make it positive and also that way don't interefere with the `cx.stop_propagation` at the bottom.

Reply:
I like the idea of wrapping all the middle-click-paste logic in a positive check of the condition, it reads much clearer.  My understanding is that we probably shouldn't stop the event from propagating if we're not doing anything with it here, is that your understanding too?  That's how it's done with this block too:
```rust
            if !text_hitbox.is_hovered(cx) || editor.read_only(cx) {
                return;
            }
```